fake_service/schema/views.py

```diff
@@ -13,5 +17,36 @@ class SchemaAPIView(ModelViewSet):
     def create(self, request, *args, **kwargs):
         serializer = self.get_serializer(data=request.data)
         serializer.is_valid(raise_exception=True)
-        create_schema(data=request.data, user=request.user)
-        return Response(status=status.HTTP_201_CREATED)
+        response = SchemaCreate(data=request.data, user=request.user).create_schema
+        return Response(response)
+
+    @action(detail=True, methods=["post"])
+    def generate_data(self, request, pk) -> Response:
+        count = request.query_params.get("count", 1)
+        cache_task_key = start_download_dataset_task(user_id=request.user.id, schema_id=pk, count=count)
+        return Response({"Cache task key is": f"{cache_task_key}"}, status=status.HTTP_201_CREATED)
```

Comment:
bad naming for "key" in response. Just do `{"cache_key": cache_task_key} `. Also i think Http 202 whould be better status code for this particular case

Reply:
Also the naming is pretty bad. Your view says `generate_data`, but you call `start_download_dataset_task`. Its confusing